cyences_app_for_splunk/bin/device_inventory_v2_util.py

```diff
@@ -49,349 +114,425 @@ def _internal_check_list_field_format(self, field):
+        if self.modify_devices:
+            self._convert_dict_to_str()
+            # splunk.BadRequest: [HTTP 400] Bad Request; [{'type': 'ERROR', 'code': None, 'text': 'Request exceeds API limits - see limits.conf for details. (Too many documents for a single batch save, max_documents_per_batch_save=1000)'}]
+            if len(self.modify_devices) < 1000:
+                jsonargs = json.dumps(self.modify_devices)
+                _ = rest.simpleRequest(
+                    "/servicesNS/nobody/{}/storage/collections/data/{}/batch_save?output_mode=json".format(
+                        cs_utils.APP_NAME, collection_name
+                    ),
+                    method="POST",
+                    jsonargs=jsonargs,
+                    sessionKey=self.session_key,
+                    raiseAllErrors=True,
+                )
+            else:
+                updated_data_full = [
+                    self.modify_devices[i : i + 800]
+                    for i in range(0, len(self.modify_devices), 800)
+                ]  # send max 800 in each chunk
+                for chunk in updated_data_full:
+                    jsonargs = json.dumps(chunk)
+                    _ = rest.simpleRequest(
+                        "/servicesNS/nobody/{}/storage/collections/data/{}/batch_save?output_mode=json".format(
+                            cs_utils.APP_NAME, collection_name
+                        ),
+                        method="POST",
+                        jsonargs=jsonargs,
+                        sessionKey=self.session_key,
+                        raiseAllErrors=True,
+                    )
+            self.logger.info(
+                "Updated {} entries in the lookup.".format(len(self.modify_devices))
+            )
         else:
-            values2_updated = [element.lower() for element in values2]
-            current_list2_updated = [element.lower() for element in current_list2]
-
-        for val1 in values1_updated:
-            for existing_val1 in current_list1_updated:
-                if val1 == existing_val1:
-                    for val2 in values2_updated:
-                        if val2 in current_list2_updated:
-                            return True
-        return False
+            self.logger.info("No entries to update in the KVStore lookup.")
+
+    def delete_kvstore_entries(self, collection_name):
+        for dvc in self.modify_devices:
+            _ = rest.simpleRequest(
+                "/servicesNS/nobody/{}/storage/collections/data/{}/{}".format(
+                    cs_utils.APP_NAME, collection_name, dvc.get("_key")
+                ),
+                method="DELETE",
+                sessionKey=self.session_key,
+                raiseAllErrors=True,
+            )
 
+    def __enter__(self):
+        return self
 
-    def is_match(self, device_entry, hostname_postfixes=[]):
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.operation == "cleanup":
```

Comment:
Keep two list:
* updated_devices
* deleted_devices

Perform operation on both lists on all the exits.
That way we don't need self.operation

Reply:
Before updated_devices execution, remove entries from the updated_devices list that are also part of deleted_devices.

Reply:
Done